src/mapping-v2-core.ts

```diff
@@ -288,12 +295,13 @@ export function handleExternalAssetDependencyRemoved(
 
 export function handleGatewayUpdated(event: GatewayUpdated): void {
   let contractEntity = Contract.load(event.address.toHexString());
-  
+
   if (!contractEntity) {
     log.warning("Contract not found for GatewayUpdated event", []);
     return;
   }
-  const dependencyType = FLEX_CONTRACT_EXTERNAL_ASSET_DEP_TYPES[event.params._dependencyType];
+  const dependencyType =
+    FLEX_CONTRACT_EXTERNAL_ASSET_DEP_TYPES[event.params._dependencyType];
   if (dependencyType === "IPFS") {
     contractEntity.preferredIPFSGateway = event.params._gatewayAddress;
   } else {
```

Comment:
Do we need to clear the dependency not being set (e.g. clear arweave if setting IPFS)?

Reply:
a project can have assets from both IPFS and Arweave, so I don't think we'd want to clear

Reply:
ahhhh makes sense! SGTM!